docs/cmdline-opts/append.d

```diff
@@ -2,6 +2,7 @@ Short: a
 Long: append
 Help: Append to target file when uploading
 Protocols: FTP SFTP
+Category: ftp sftp
```

Comment:
I would say `categories`
```suggestion
Categories: ftp sftp
```
I see that in most cases, `categories` is equal to `protocols`. Does it make sense to use by default the protocols field as the categories and in specific cases use the `categories` as an override field ?

Reply:
Thanks for your review!
Most options only belong to one category. The plural would sound weird there in my opinion.

Reply:
> Does it make sense to use by default the protocols field as the categories

That sounds like a lever idea that will reduce a lot of duplication!

Reply:
> That sounds like a lever idea that will reduce a lot of duplication!

But what is with options like `--user` or `--output` which don't belong to any (specific) protocol

Reply:
> Most options only belong to one category. The plural would sound weird there in my opinion.

In your code you are already referencing `categories`:
https://github.com/curl/curl/blob/af4dd100c6da95af1c3491de24a4a04f43326c1f/src/tool_help.c#L119-L123

Protocols is in plural form and b/c category(ies) is relatively equal to protocols I think we need to stay consistent.

> But what is with options like --user or --output which don't belong to any (specific) protocol

We could you use this logic to define the category(ies):
```yaml
my_category(ies): category(ies) | default(protocols) | default("curl")
```
What do you think?

Reply:
Can you explain the syntax? I don't understand

Reply:
> Can you explain the syntax? I don't understand

`categories` would be equal to `categories` if it exists in the file, if not using the `protocols` value, if not using `"curl"` which would be the very last default value.

Reply:
I actually disagree with that suggestion. Of course it reduces some work but it also makes stuff more unclear in my opinion

Reply:
> I actually disagree with that suggestion. Of course it reduces some work but it also makes stuff more unclear in my opinion

What makes stuff unclear? using by default `"curl"` if neither `categories` or `protocols` exists?

Reply:
I'm not so sure about the default "curl" category - probably because I think the `Category:` (or `ies`) header should be present in all files just to make it explicit that the author has considered them, but I think using the protocols as implicit categories seems like a way to avoid duplication and over all just make sure they match.

Reply:
@bagder I'm not sure to understand you point of view :thinking: ; tell me if I am wrong:
1. each file has a `category:` field
2. `and over all just make sure they match`, so you have to duplicate the protocols in the categories to make sure it is correct?

Reply:
Yes I think:

- each .d file gets a `category:`  field, that can be empty or list *additional* categories apart from the protocols which are used by default
- The category `curl` should rather be renamed `tool` and should be explicitly mentioned as a category if desired.

This way, the used categories are all visible in each file (just that they're also in the protocols line) and we don't need to repeat the protocols in the category field.

Reply:
Sounds good to me, your position @emilengler :smiley: ?

Reply:
I will have a deeper look on this at Sunday or Monday. I am currently on vacation

Reply:
The change from tool to curl makes sense. I will change this.
I also agree on the part that the category field can be empty where the item would fall into the misc category then.
However I still don't think it is a good idea to mix up with protocols. Of course this saves some work but I still think it adds more confusion also AFAIK the protocol field is not omnipresent so far